Apps/WebClient/src/ClientApp/src/store/modules/laboratory/actions.ts

```diff
@@ -86,6 +87,72 @@ export const actions: LaboratoryActions = {
             { root: true }
         );
     },
+    retrieveLaboratoryOrders(
+        context,
+        params: { hdid: string }
+    ): Promise<RequestResult<LaboratoryOrder[]>> {
+        const logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
+        const laboratoryService: ILaboratoryService =
+            container.get<ILaboratoryService>(
+                SERVICE_IDENTIFIER.LaboratoryService
+            );
+
+        return new Promise((resolve, reject) => {
+            const laboratoryOrders: LaboratoryOrder[] =
+                context.getters.laboratoryOrders;
+            if (context.state.authenticated.status === LoadStatus.LOADED) {
+                logger.debug("Laboratory Orders found stored, not querying!");
+                resolve({
+                    pageIndex: 0,
+                    pageSize: 0,
+                    resourcePayload: laboratoryOrders,
+                    resultStatus: ResultType.Success,
+                    totalResultCount: laboratoryOrders.length,
+                });
+            } else {
+                logger.debug("Retrieving Laboratory Orders");
+                context.commit("setLaboratoryOrdersRequested");
+                laboratoryService
+                    .getLaboratoryOrders(params.hdid)
+                    .then((result) => {
+                        if (result.resultStatus === ResultType.Success) {
+                            EventTracker.loadData(
+                                // Laboratory and COVID-19 shares the same modules
+                                EntryType.Covid19LaboratoryOrder,
```

Comment:
There should be a separate entry type for LaboratoryOrder.  Suggestion: `LaboratoryOrder = "AllLaboratory",`